### DISCUSSION TITLE
Referencing columns of tables via enums for dynamic access

### DISCUSSION THREAD

**Discussion by ChristianIvicevic** [prisma/prisma#14789] [Q&A]
I've got a question regarding mapping of different representations of names. In my game every player can construct a variety of buildings as such my database model looks somewhat like this
```
model Planet {
  // ...
  solarPlant Int
  metalMine Int
  // ...
}
```
The values stored indicate the current level of a building and I did add certain events to the game that sometimes reference those, for example finishing the construction of a building and thus increasing its level. The simplified model for those events looks as follows
```
model ConstructionEvent {
  // ...
  building Building
  // ...
}
enum Building {
  SOLAR_PLANT
  METAL_MINE
  // ...
}
```
Here comes the fun part! Sometimes I query for the buildings on a planet and want to dynamically access its values which may look like this
```typescript
const event = prisma.constructionEvent.findUnique({/* ... */});
const planet = prisma.planet.findUnique({/* ... */});

const building2Level = planet[event.building]; // Incompatible - TypeScript even complains
prisma.planet.update({
  data: {
    // Incompatible - TypeScript doesn't complain...
    [event.building]: { increment: 1 }
  }
});
```
To make this compatible with one-another I'd require at least one, maybe two, `Record`s to map `Building` to `Pick<Planet, 'solarPlant' | 'metalMine' | /* ... */>` and vice-versa which isn't really difficult, just annoying that I'd need to use those records whenever attempting to index as shown.

One other option I was just thinking of was to name the columns in the `Planet` model exactly like the enum values, so
```
model Planet {
  // ...
  SOLAR_PLANT Int
  METAL_MINE Int
  // ...
}
```
This looks super ugly, but given that I'd very rarely access these fields directly via `planet.SOLAR_PLAN` and more often dynamically as shown in the pevious snippets this might make the mappers redundant.

Is there a more suitable way of building the models here or would you stick to the mapping within TypeScript-land as described?

**Comment by pantharshit00** [comment]
Hello @ChristianIvicevic , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

If this is still relevant, I would suggest you to may try mapping the enum values to the same name to make them compatible:
```prisma
enum Building {
  solarPlant @map("SOLAR_PLANT")
  metalMine @map("METAL_MINE")
  // ...
}
```

That will not result in a typescript error and everything will be typesafe.

**Reply by pantharshit00** [reply]
Hello @ChristianIvicevic 

I am going through our past discussions right now and I want to make sure this is not an issue for you. Can you please confirm if this is still relevant to you?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
